src/plugins/kubectlGeneration/aksDocsRag/client.ts

```diff
@@ -0,0 +1,122 @@
+
+export function getAKSDocsRAGClient(sessionProvider: ReadyAzureSessionProvider) {
+    return new AKSDocsRAGClient(sessionProvider);
+}
+
+type RequestConfig = {
+    message: string;
+};
+
+export interface CommandResponse {
+    status: string;
+    message: string;
+    code: string;
+}
+
+export class AKSDocsRAGClient {
+    private authToken: string;
+    private tokenExpiresAt: number;
+    private sessionProvider: ReadyAzureSessionProvider;
+    private authScopes: string[];
+
+    constructor(sessionProvider: ReadyAzureSessionProvider) {
+        this.authToken = "";
+        this.tokenExpiresAt = 0;
+        this.sessionProvider = sessionProvider
+        this.authScopes = aksDocsRAGScopes;
+    }
+
+    private async fetchToken() : Promise<void> {
+        try {
+            const token = await this.sessionProvider.getAuthSession({ scopes: this.authScopes });
+
+            if (failed(token)) {
+                throw new Error(`No Microsoft authentication session found: ${token?.error}`);
+            }
+
+            this.authToken = token.result.accessToken;
+        } catch (error) {
+            throw new Error(`Failed to fetch token: ${getErrorMessage(error)}`);
+        }
+    }
+
+    private async refreshToken() : Promise<void> {
+        try {
+            if (!this.authToken || this.tokenExpiresAt < Date.now()) {
+            await this.fetchToken();
+            }
+        } catch (error) {
+            throw new Error(`Failed to refresh token: ${getErrorMessage(error)}`);
+        }
+    }
+
+    public async sendRequest(config: RequestConfig): Promise<Errorable<{ response: CommandResponse }>> {
+        await this.refreshToken();
+        const { message } = config;
+
+        let rawResponse = undefined;
+        try {
+            rawResponse = await fetch(aksDocsRAGEndpoint, {
+                method: "POST",
+                headers: {
+                    "Access-Control-Allow-Origin": "*",
+                    Authorization: `Bearer ${this.authToken}`,
+                    "App-Tenant-Id": this.sessionProvider.selectedTenant.id,
+                },
+                body: JSON.stringify({
+                    IsLocalEvaluation: false,
+                    Question: message,
+                    Scenario: aksDocsRAGScenario,
+                    Intent: aksDocsRAGIntent,
+                }),
+            });
+        } catch (err: unknown) {
+            return { succeeded: false, error: `Unable to fetch data from AKS Docs RAG endpoint: ${getErrorMessage(err)}` };
+        }
+
+        if (!rawResponse) {
+            return { succeeded: false, error: `Unable to fetch data from AKS Docs RAG endpoint` };
+        }
+
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const data: any = await rawResponse.json();
+        const status = rawResponse.status;
+
+        if (status === 424) {
```

Comment:
💡 idea: MIght b add as code level comment with document of these status code, also if it make sense please think of renaming the variable to `statusCode` or more logically closer to real intent.

Reply:
This documentation exists in azure ML promptflow, I believe. Which is not public to users

Reply:
Sounds good, might be just write like this `// this status code means <something relevant meaning>` please.

Reply:
Thanks Tats, updated comment